### PR TITLE
audit

### PR DESCRIPTION
**Pull request by cbadawi**

### CODE REVIEW

**Comment by cbadawi** [comment]
@r0zar  only issue is the JSON.stringify in message.tsx that is adding the JSON to the message
Screenshot 2024-11-24 at 1 33 58 AM

dont know why its needed / whether to remove it
https://github.com/pointblankdev/intelligence-web/blob/feat/audit/components/custom/message.tsx

File: tools/code-audit/smart-contract.ts [hunk 1]
```diff
@@ -26,18 +27,23 @@ const contractAuditParamsSchema = z.object({
...
 
 type ContractAuditResponse = {
   success: boolean;
-  data?: {
-    fungibleTokens: Array<{
-      tokenIdentifier: string;
-    }>;
-  };
+  data?:
+    | {
```
Comment (r0zar) [inline tools/code-audit/smart-contract.ts:39]: Is this an OR operation?

**Review by r0zar — COMMENTED** [review]
Commented

File: tools/code-audit/smart-contract.ts (hunk 1 above)
Comment (cbadawi) [reply tools/code-audit/smart-contract.ts:39]: yea but it can be written clearer, i'll update

Comment (cbadawi) [reply tools/code-audit/smart-contract.ts:39]: @r0zar fixed

Comment (r0zar) [reply tools/code-audit/smart-contract.ts:39]: Okay. i only ask because we need that tokenIdentifier to use in post-condition contract calls from the web app. thats actually the reason i made it at all, then decided to expand it to a full audit tool since i was already scanning the contract source code. but in retrospect they should be decoupled. 

TLDR; the charisma app looks for the tokenIdentifier at that property, and if its not there those tokens will fail swaps.

Comment (cbadawi) [reply tools/code-audit/smart-contract.ts:39]: ah ok, was wondering why it was there. yea i didnt break anything but would make sense to separate them now/later.

Comment (r0zar) [reply tools/code-audit/smart-contract.ts:39]: 👍

**Review by r0zar — APPROVED** [review]
LGTM 👍
